content/_code-samples/key-derivation/js/index.js

```diff
@@ -0,0 +1,219 @@
+'use strict';
+
+//organize imports
+const assert = require("assert")
+const brorand = require("brorand")
+const BN = require("bn.js")
+const elliptic = require("elliptic");
+const Ed25519 = elliptic.eddsa('ed25519');
+const Secp256k1 = elliptic.ec('secp256k1');
+const { rfc1751MnemonicToKey, keyToRFC1751Mnemonic } = require("xrpl/dist/npm/Wallet/rfc1751");
+const hashjs = require("hash.js");
+const Sha512 = require("ripple-keypairs/dist/Sha512")
+const { codec, encodeAccountPublic, encodeNodePublic } = require("ripple-address-codec");
+
+const XRPL_SEED_PREFIX = 0x21
+
+const isHex = function(value) {
+    const regex = new RegExp(/^0x([0-9a-f]+(\.[0-9a-f]*)?|\.[0-9a-f]+)(p[+-]?\d+)?$/, 'i')
+    return regex.test(value)
+}
+
+const bytesToHex = function(bytes) {
+    return Array.from(bytes, (byteValue) => {
+        const hex = byteValue.toString(16).toUpperCase();
+        return hex.length > 1 ? hex : `0${hex}`;
+    }).join('');
+}
+
+const hexToBytes = function(hex) {
+    assert.ok(hex.length % 2 === 0);
+    return hex.length === 0 ? [] : new BN(hex, 16).toArray(null, hex.length / 2);
+}
+
+const encodeUTF8 = function (string) {
+    return encodeURIComponent(string)
+}
+
+const sha512half = function (value) {
+    return hashjs.sha512().update(value).digest().slice(0, 32);
+}
+
+class Seed {
+    
+    constructor(seedValue = '', checkRFC1751 = false) {
+        this.bytes = this._initSeed(seedValue)
+        this.ED25519Keypair = this._deriveED25519Keypair(this.bytes)
+        this.Secp256K1Keypair = this._deriveSecp256K1Keypair(this.bytes)
+    }
+
+    _initSeed(seedValue) {
+
+        //default to random seed
```

Comment:
Style note - please add a space and capitalize the first letter in comments
```suggestion
        // Default to random seed
```

Reply:
I've updated all comments according to styleguide